deps/picotls/lib/openssl.c

```diff
@@ -693,77 +695,138 @@ int ptls_openssl_create_key_exchange(ptls_key_exchange_context_t **ctx, EVP_PKEY
     }
 }
 
+struct sign_ctx {
+    const struct st_ptls_openssl_signature_scheme_t *scheme;
+    EVP_MD_CTX *ctx;
+    ptls_buffer_t buf;
+    size_t siglen;
```

Comment:
I wonder if we can do something as simple as:

```c
struct sign_ctx {
    ...
    size_t siglen;
    uint8_t sig[0];
};
...
sign_ctx = malloc(offsetof(*sign_ctx) + siglen);
```

Looking at the code, to me it seems that we can instantiate `sign_ctx` after determining the length.